Sources/Core/Renderer/GUI/GUIElement/GUITextInput.swift

```diff
@@ -12,21 +13,46 @@ struct GUITextInput: GUIElement {
     ).meshes(context: context)
 
     // Message
-    var message = try content.meshes(context: context)
-    message.translate(amount: [2, 2])
+    var firstMessage = try String(content.prefix(upTo: content.index(content.endIndex, offsetBy: -cursorIndex))).meshes(context: context)
```

Comment:
Separate the string splitting from the mesh generation just to make this code a bit more readable. And it'd make sense to rename these variables as well because the current names kinda make it sound like there are two messages, whereas it's actually just two parts of the same message.

```swift
var messageBeforeCursor
var messageAfterCursor
var messageMeshBeforeCursor
var messageMeshAfterCursor
```

Something like that maybe?